### PR TITLE
Plugin permission bug fix

### PR DESCRIPTION
**Pull request by vln104**
Previously, permissions created by plugins would cause an error when saved because generatePermissions function did not set pluginBundle flag properly when calling getPermissionObject.

To reproduce the issue: 
- Use the helloworld plugin included the [Developer document](https://developer.mautic.org/#creating-custom-permissions).
- Install the plugin and navigate to Roles -> Add new role
- Go to your plugin's permission and select some the click save
- An error page should appear.

To verify fix:
- Apply the changes in this PR and repeat above process. There should no longer be an error after saving permissions. 

Link to related forum topic, including the stack trace after the error is encountered: https://www.mautic.org/community/index.php//1042-custom-permission-plugin

### CODE REVIEW

**Comment by escopecz** [comment]
@vln104 would you be able to provide a plugin for testing of this PR? The testing would take a lot longer if every tester would have to build his own plugin with the permission functionality.

**Comment by vln104** [comment]
yes I have a hello world bundle that can be used to test:

https://www.dropbox.com/s/hi9p3sk1gfzstdj/HelloWorldBundle.zip?dl=0

**Comment by escopecz** [comment]
1) The HelloWorld plugin has a small bug:
500 Internal Server Error - Case mismatch between loaded and declared class names: MauticPlugin\HelloWorldBundle\Security\Permissions\HelloWorldPermissions vs MauticPlugin\HelloworldBundle\Security\Permissions\HelloWorldPermissions
It's fixed here: [MauticCloudStorageBundle.zip](https://github.com/mautic/mautic/files/59571/MauticCloudStorageBundle.zip)

2) The PR has a git conflict. Could you pull staging into its branch, resolve the conflict and push again?

Other than that, the PR works :+1:

**Comment by vln104** [comment]
I pulled staging into my branch and there are lots of changes. Do I just commit the change I initially made for this bug fix? Or do I commit everything, including other files that have been changed due to the merge but not made by me initially?

**Comment by alanhartless** [comment]
@vln104 When you pull, merge and push the changes to your branch, git will auto-handle the rest. It knows what is already included in staging and will only include your changes in this PR.

**Comment by vln104** [comment]
Thanks @alanhartless. I've resolved and committed the changes. 

Git seems much better in svn in working out changes :)

File: app/bundles/CoreBundle/Security/Permissions/CorePermissions.php [hunk 1]
```diff
@@ -215,7 +215,8 @@ public function generatePermissions(array $permissions)
                 $entity->setName(strtolower($name));
 
                 $bit   = 0;
-                $class = $this->getPermissionObject($bundle);
+                $pluginBundles = $this->getPluginBundles();
```
Comment (alanhartless) [inline app/bundles/CoreBundle/Security/Permissions/CorePermissions.php]: Can you move this to be outside the foreach loops? Somewhere around line 207 or 208. Not sure it needs to be fetched on each foreach iteration.

**Comment by vln104** [comment]
@alanhartless that is a good suggestion. I have done as suggested and the external plugin's permission could still be saved for me after the change.

**Comment by alanhartless** [comment]
:+1:

**Pull request merged**